src/common/float4.cpp

```diff
@@ -97,12 +97,12 @@ uint8_t float2e3m0(float f) {
 
     // There is no NaN or infinity in e3m0, we just return maxval
     uint32_t naninf_mask = 0x7f800000;
-    if ((f_raw & naninf_mask) == naninf_mask) return 0x7;
+    if ((f_raw & naninf_mask) == naninf_mask) return 0x7 | (sign >> 28);
```

Comment:
Since you are updating this, is there any particular difference between E2M1 (where we return 0x0 - the minimum value) and E3M0 (where we return 0x7 - the maximum value) handling for infinity/NaNs? Shouldn't they both return min or max?

Reply:
According to [MXFP spec](https://www.opencompute.org/documents/ocp-microscaling-formats-mx-v1-0-spec-final-pdf):
- inf is an out-of-range value and should be clamped to max value
- for NaN it is implementation defined.

Hence I would suggest to actually modify the e2m1 impl to return the max value instead of 0.